File: app/src/main/java/org/schulcloud/mobile/data/model/responseBodies/SignedUrlResponse.java
package org.schulcloud.mobile.data.model.responseBodies;

import com.google.gson.annotations.SerializedName;

public class SignedUrlResponse {
    public String url;
    public SignedUrlResponseHeader header;


    public static class SignedUrlResponseHeader {
        @SerializedName("Content-Type")
        private String contentType;

        @SerializedName("x-amz-meta-path")
        private String metaPath;

        @SerializedName("x-amz-meta-name")
        private String metaName;

        @SerializedName("x-amz-meta-flat-name")
        private String metaFlatName;

        @SerializedName("x-amz-meta-thumbnail")
        private String metaThumbnail;

        public SignedUrlResponseHeader() {

        }

        public String getContentType() {
            return contentType;
        }
        public void setContentType(String contentType) {
            this.contentType = contentType;
        }

        public String getMetaPath() {
            return metaPath;
        }
        public void setMetaPath(String metaPath) {
            this.metaPath = metaPath;
        }

        public String getMetaName() {
            return metaName;
        }
        public void setMetaName(String metaName) {
            this.metaName = metaName;
        }

        public String getMetaFlatName() {
            return metaFlatName;
        }
        public void setMetaFlatName(String metaFlatName) {
            this.metaFlatName = metaFlatName;
        }

        public String getMetaThumbnail() {
            return metaThumbnail;
        }
        public void setMetaThumbnail(String metaThumbnail) {
            this.metaThumbnail = metaThumbnail;
        }
    }
}

File: app/src/main/java/org/schulcloud/mobile/models/file/SignedUrlResponse.kt
package org.schulcloud.mobile.models.file

import com.google.gson.annotations.SerializedName


class SignedUrlResponse {
    var url: String? = null
    var header: SignedUrlResponseHeader? = null


    class SignedUrlResponseHeader {
        @SerializedName("Content-Type")
        var contentType: String? = null

        @SerializedName("x-amz-meta-path")
        var metaPath: String? = null

        @SerializedName("x-amz-meta-name")
        var metaName: String? = null

        @SerializedName("x-amz-meta-flat-name")
        var metaFlatName: String? = null

        @SerializedName("x-amz-meta-thumbnail")
        var metaThumbnail: String? = null
    }
}
